packages/lockfile-file/src/experiments/inlineSpecifiersLockfileConverters.ts

```diff
@@ -0,0 +1,105 @@
+import type { Lockfile, ProjectSnapshot, ResolvedDependencies } from '@pnpm/lockfile-types'
+import type { InlineSpecifiersLockfile, InlineSpecifiersProjectSnapshot, InlineSpecifiersResolvedDependencies } from './InlineSpecifiersLockfile'
+
+export function isExperimentalInlineSpecifiersFormat (
+  lockfile: InlineSpecifiersLockfile | Lockfile
+): lockfile is InlineSpecifiersLockfile {
+  return (lockfile as InlineSpecifiersLockfile).inlineSpecifiersFormat ?? false
+}
+
+export function convertToInlineSpecifiersFormat (lockfile: Lockfile): InlineSpecifiersLockfile {
+  return {
+    inlineSpecifiersFormat: true,
+    ...lockfile,
+    importers: mapValues(lockfile.importers, convertProjectSnapshotToInlineSpecifiersFormat),
+  }
+}
+
+export function revertFromInlineSpecifiersFormatIfNecessary (lockfile: Lockfile | InlineSpecifiersLockfile): Lockfile {
+  return isExperimentalInlineSpecifiersFormat(lockfile)
+    ? revertFromInlineSpecifiersFormat(lockfile)
+    : lockfile
+}
+
+export function revertFromInlineSpecifiersFormat (lockfile: InlineSpecifiersLockfile): Lockfile {
+  const { importers, inlineSpecifiersFormat, ...rest } = lockfile
+
+  return {
+    ...rest,
+    importers: mapValues(importers, revertProjectSnapshot),
+  }
+}
+
+function convertProjectSnapshotToInlineSpecifiersFormat (
+  projectSnapshot: ProjectSnapshot
+): InlineSpecifiersProjectSnapshot {
+  const { specifiers } = projectSnapshot
+  return {
+    dependencies: projectSnapshot.dependencies != null
+      ? convertResolvedDependenciesToInlineSpecifiersFormat(projectSnapshot.dependencies, { specifiers })
```

Comment:
it will be shorter if you add specifiers as the first argument and then make a new function. like

```
const _convertResolvedDependenciesToInlineSpecifiersFormat = convertResolvedDependenciesToInlineSpecifiersFormat.bind(null, { specifiers })
```

Reply:
or you may use a loop

```
for (const depField of DEPENDENCIES_FIELDS) {
  newImporter[depField] = projectSnapshot[depField]......
}
```

Reply:
Agree this could have been much shorter. How does the functional approach that was just pushed look?

https://github.com/pnpm/pnpm/blob/3cd2784df62f6bfc454460367424f96e6b0e27f2/packages/lockfile-file/src/experiments/inlineSpecifiersLockfileConverters.ts#L50-L58